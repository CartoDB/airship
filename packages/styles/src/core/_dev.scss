@import 'tools/_mixins';
@import 'variables/_variables';
