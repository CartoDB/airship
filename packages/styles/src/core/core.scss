@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
