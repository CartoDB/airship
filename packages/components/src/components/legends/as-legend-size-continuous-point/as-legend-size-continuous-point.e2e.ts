import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-legend-size-continuous-point', async () => {
  describe('Rendering', async () => {
    let page: E2EPage;

    it('basic', async () => {
      page = await newE2EPage({ html: `
        <as-legend-size-continuous-point>
        </as-legend-size-continuous-point>
      ` });

      const element = await page.find('as-legend-size-continuous-point');
      element.setProperty('data', [{
        label: 'point label',
        width: 10,
      },
      {
        label: 'line label',
        width: 12,
      },
      {
        label: 'polygon label',
        width: 14,
      }]);
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('basic horizontal', async () => {
      page = await newE2EPage({ html: `
        <as-legend-size-continuous-point>
        </as-legend-size-continuous-point>
      ` });

      const element = await page.find('as-legend-size-continuous-point');
      element.setProperty('data', [{
        label: 'point label',
        width: 10,
      },
      {
        label: 'line label',
        width: 12,
      },
      {
        label: 'polygon label',
        width: 14,
      }]);
      element.setProperty('orientation', 'horizontal');
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
