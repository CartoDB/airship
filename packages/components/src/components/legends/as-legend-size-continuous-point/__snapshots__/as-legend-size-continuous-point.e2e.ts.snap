// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-legend-size-continuous-point Rendering basic 1`] = `"<as-legend-size-continuous-point class=\\"hydrated\\"> <div class=\\"as-legend-size-continuous-point--wrapper as-legend-size-continuous-point--vertical\\"><span class=\\"as-legend-size-continuous-point--label\\">point label</span><div class=\\"as-legend-size-continuous-point--steps\\" style=\\"height: 14px; width: 14px;\\"><div class=\\"as-legend-size-continuous-point--circle\\" style=\\"height: 16px; width: 16px;\\"></div><div class=\\"as-legend-size-continuous-point--circle\\" style=\\"height: 14px; width: 14px;\\"></div><div class=\\"as-legend-size-continuous-point--circle\\" style=\\"height: 12px; width: 12px;\\"></div></div><span class=\\"as-legend-size-continuous-point--label\\">polygon label</span></div></as-legend-size-continuous-point>"`;

exports[`as-legend-size-continuous-point Rendering basic horizontal 1`] = `"<as-legend-size-continuous-point class=\\"hydrated\\"> <div class=\\"as-legend-size-continuous-point--wrapper as-legend-size-continuous-point--horizontal\\"><span class=\\"as-legend-size-continuous-point--label\\">point label</span><div class=\\"as-legend-size-continuous-point--steps\\" style=\\"height: 14px; width: 14px;\\"><div class=\\"as-legend-size-continuous-point--circle\\" style=\\"height: 16px; width: 16px;\\"></div><div class=\\"as-legend-size-continuous-point--circle\\" style=\\"height: 14px; width: 14px;\\"></div><div class=\\"as-legend-size-continuous-point--circle\\" style=\\"height: 12px; width: 12px;\\"></div></div><span class=\\"as-legend-size-continuous-point--label\\">polygon label</span></div></as-legend-size-continuous-point>"`;
